src/wifes_calib.py

```diff
@@ -793,36 +793,55 @@ def derive_wifes_telluric(cube_fn_list,
             ex_data = numpy.loadtxt(extract_in_list[i])
             obs_wave = ex_data[:,0]
             obs_flux = ex_data[:,1]
+        
+        # Filtering nan values in case of missing flux values
+
+        filter_nan = ~numpy.isnan(obs_flux)
+        obs_wave_nan = obs_wave[filter_nan]  
+        obs_flux_nan = obs_flux[filter_nan]
+        
+
+        print("OBS_WAVE_NAN", obs_wave_nan)
+        print("OBS_FLUX", obs_flux_nan)
+        print("O2_tell_bands", O2_telluric_bands)
+        print("H2O_tell_bands", H2O_telluric_bands)
+
+
+
         # define all the telluric regions
-        O2_mask  = wavelength_mask(obs_wave, O2_telluric_bands)
-        H2O_mask = wavelength_mask(obs_wave, H2O_telluric_bands)
+        O2_mask  = wavelength_mask(obs_wave_nan, O2_telluric_bands)
+        H2O_mask = wavelength_mask(obs_wave_nan, H2O_telluric_bands)
         O2_inds  = numpy.nonzero(O2_mask==0)[0]
         H2O_inds = numpy.nonzero(H2O_mask==0)[0]
         # fit smooth polynomial to non-telluric regions!
         fit_inds = numpy.nonzero(O2_mask*H2O_mask
-                                 *(obs_wave >= fit_wmin)
-                                 *(obs_wave <= fit_wmax))[0]
-        smooth_poly = numpy.polyfit(obs_wave[fit_inds],
-                                    obs_flux[fit_inds],
+                                 *(obs_wave_nan >= fit_wmin)
+                                 *(obs_wave_nan <= fit_wmax))[0]
+        
+        print("OBS_INDS",fit_inds)
```

Comment:
As above, remove debug? If it's generally useful, let me know.

Reply:
Yes, it's removed now. Thank you!